app/assets/src/components/ui/controls/SearchBox.jsx

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { Search } from "semantic-ui-react";
+
+class SearchBox extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.source = this.props.source;
+
+    this.resetComponent = this.resetComponent.bind(this);
+    this.handleSearchChange = this.handleSearchChange.bind(this);
+    this.handleResultSelect = this.handleResultSelect.bind(this);
+  }
+
+  componentWillMount() {
+    this.resetComponent();
+  }
+
+  resetComponent() {
+    this.setState({ isLoading: false, results: [], value: "" });
+  }
+
+  handleResultSelect(e, { result }) {
+    this.setState({ value: result.title });
+    this.props.onResultSelect(e, { result });
+  }
+
+  handleSearchChange(e, { value }) {
+    this.setState({ isLoading: true, value });
+
+    setTimeout(() => {
+      if (this.state.value.length < 1) return this.resetComponent();
+
+      const re = new RegExp(_.escapeRegExp(this.state.value), "i");
+      const isMatch = result => re.test(result.title);
+
+      this.setState({
+        isLoading: false,
+        results: _.filter(this.source, isMatch)
```

Comment:
Do we have to use lodash functions? Just trying to keep some consistency.
This, for instance, can just be this.source.filter(isMatch). Not sure how easy it is to replace the rest.

Reply:
ok replaced this instance and will look into others